### PR TITLE
ci: Add trap to remove tmp files

### LINKED ISSUE #1613: trap cleanup of tmp files in static check script
See: https://github.com/kata-containers/tests/pull/1612#discussion_r286393094.

Ideally, the trap handler would:

- Delete the logs on success
- Display a list of logs on failure (but not delete them).

### PR DESCRIPTION
**Pull request by GabyCT**
This adds trap cleanup for the temporary files
used in static checks.

Fixes #1613

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: .ci/static-checks.sh [hunk 1]
```diff
@@ -110,6 +110,13 @@ Examples:
 EOT
 }
 
+function remove_tmp_files() {
+	rm -f "$url_map" "$invalid_urls" "$md_links" "$submodule_packages" "$all_packages" "${tmp}"
```
Comment (devimc) [inline .ci/static-checks.sh]: I think this won't work, since all these variables are local variables

Comment (jodh-intel) [reply .ci/static-checks.sh]: Yes. I think you'd need to do something like:

```sh
# List of files to delete on exit
files_to_remove=()

cleanup()
{
    rm -f ${files_to_remove[@]}
}

foo()
{
    files_to_remove+=(foo.log)

    # ...
}

bar()
{
    files_to_remove+=(bar.log)

    # ...
}

main ()
{
    trap cleanup EXIT

    # ...
}

main "$@"
```

**Pull request merged**